sdks/python/apache_beam/utils/thread_pool_executor.py

```diff
@@ -134,3 +134,12 @@ def shutdown(self, wait=True):
       if wait:
         for worker in self._workers:
           worker.join()
+
+
+class _SharedUnboundedThreadPoolExecutor(UnboundedThreadPoolExecutor):
+  def shutdown(self, wait=True):
+    # Prevent shutting down the shared thread pool
+    pass
+
+
+SharedUnboundedThreadPoolExecutor = _SharedUnboundedThreadPoolExecutor()
```

Comment:
Maybe give it a constant name? `SHARED_THREAD_POOL_EXECUTOR` or something like that? To clarify it's an instance, and not a class.

I know we did not do this for `MetricsEnvironment`, but I think it may be better.

Reply:
+1, this is confusing.
I would declare the global instance private to ```thread_pool_executor``` module (```_shared_thread_pool_executor```), and add a public getter function: ```thread_pool_executor.get_shared_executor()```.

This is also recommended by http://google.github.io/styleguide/pyguide.html#254-decision.

Reply:
done